tests/modules/programs/abook/with-settings.nix

```diff
@@ -0,0 +1,36 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  config = {
+    programs.abook = {
+      enable = true;
+
+      extraConfig = ''
+        #  Defining a new custom field
+        # -----------------------------
+
+        field pager = Pager
+        field address_lines = Address, list
+        field birthday = Birthday, date
+
+        #  Defining a view/tab
+        # ---------------------
+
+        view CONTACT = name, email
+        view ADDRESS = address_lines, city, state, zip, country
+        view PHONE = phone, workphone, pager, mobile, fax
+        view OTHER = url, birthday
+
+        # Automatically save database on exit
+        set autosave=true
+      '';
+    };
+
+    nmt.script = ''
+      assertFileExists home-files/.config/abook/abookrc
+      assertFileContent home-files/.config/abook/abookrc ${./expected.cfg} 
```

Comment:
Can rename the `expected.cfg` file to `with-settings.cfg` to match the test file.